contact_info_service.py

```
@@ -113,6 +113,39 @@ def __get_user_prompt(self, business_name: str) -> str:
+        metrics for observability and cost tracking.
+
+        Args:
+            response: The OpenAI API response object containing usage information.
+            business_name (str): The business name being looked up, used as a metric attribute.
+
+        Note:
+            Safely handles cases where usage information is not available in the response.
+            Records both tokens_per_request histogram and tokens_total counter metrics
+            with model and business_name attributes for filtering and analysis.
+        """
+        usage = getattr(response, "usage")
```

Ruff (0.14.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)